pkg/devfile/parser/content_test.go

```diff
@@ -0,0 +1,89 @@
+package parser
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestSetDevfileContent(t *testing.T) {
+
+	const (
+		TempJsonDevfilePrefix = "odo-devfile.*.json"
+		InvalidDevfilePath    = "/invalid/path"
+		InvalidDevfileContent = ":: invalid :: content"
+	)
+
+	// createTempDevfile helper creates temp devfile
+	createTempDevfile := func(t *testing.T, content []byte) (f *os.File) {
+
+		t.Helper()
+
+		// Create tempfile
+		f, err := ioutil.TempFile(os.TempDir(), TempJsonDevfilePrefix)
```

Comment:
let's use the fake package here and write the stuff in memory instead of the disk. That will speed up the testing.